### PR TITLE
🐛Source Square: Fix Duplicate Records in Payments and Refunds Streams

### LINKED ISSUE #31237: Source Square: Payments and Refunds streams are duplicated
### Connector Name

source-square

### Connector Version

1.1.2

### What step the error happened?

During the sync

### Relevant information

There's some issues on paginating payments and refunds. For those streams airbyte is returning tons of duplicates.

### Relevant log output

_No response_

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Fixes #31237

## How
*Describe the solution*

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-square/source_square/manifest.yaml [hunk 1]
```diff
@@ -74,6 +74,9 @@ definitions:
       start_time_option:
         field_name: begin_time
         inject_into: body_json
+      end_time_option:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-square/source_square/manifest.yaml:77]: can you explain how it can fix duplicate records issue?

Comment (topefolorunso) [reply airbyte-integrations/connectors/source-square/source_square/manifest.yaml:77]: @lazebnyi The `end_time` parameter was not passed to the request. This caused the API to return all the records from the `start_time` to the present time for each time slice. So values will be duplicated this way. Makes sense?

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-square/source_square/manifest.yaml:77]: Can you present running a sync with current version and yours?

Comment (topefolorunso) [reply airbyte-integrations/connectors/source-square/source_square/manifest.yaml:77]: The current version

```{"type": "LOG", "log": {"level": "INFO", "message": "Starting syncing SourceSquare"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as STARTED"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881777159.518, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "STARTED"}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: payments "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as RUNNING"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881780809.0469, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "RUNNING"}}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "created_at": "2023-10-06T00:44:56.869Z", "updated_at": "2023-10-06T00:44:58.179Z", "amount_money": {"amount": 100, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T00:44:56.976Z", "captured_at": "2023-10-06T00:44:57.050Z"}}, "location_id": "L82360QY4MG3B", "order_id": "ocxhWW5TrHQEasZ06AgJEGXgJZSZY", "processing_fee": [{"effective_at": "2023-10-06T02:44:58.000Z", "type": "INITIAL", "amount_money": {"amount": 33, "currency": "USD"}}], "total_money": {"amount": 100, "currency": "USD"}, "approved_money": {"amount": 100, "currency": "USD"}, "receipt_number": "btMa", "receipt_url": "https://squareupsandbox.com/receipt/preview/btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T00:44:56.869Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "JrHROPUWA23bC4pAUKpfUHyjLS9dyzLHnyjZGdyq8pC6o"}, "emitted_at": 1698881780808}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "created_at": "2023-10-06T01:31:54.742Z", "updated_at": "2023-10-06T01:31:56.396Z", "amount_money": {"amount": 892657, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T01:31:54.853Z", "captured_at": "2023-10-06T01:31:54.925Z"}}, "location_id": "L82360QY4MG3B", "order_id": "KjOIV2mxV5WDn1htAgCwl7skTkKZY", "processing_fee": [{"effective_at": "2023-10-06T03:31:56.000Z", "type": "INITIAL", "amount_money": {"amount": 25917, "currency": "USD"}}], "customer_id": "7879697", "total_money": {"amount": 892657, "currency": "USD"}, "approved_money": {"amount": 892657, "currency": "USD"}, "receipt_number": "tmKJ", "receipt_url": "https://squareupsandbox.com/receipt/preview/tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T01:31:54.742Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "BOxzRrA8O6eECUPcr4Q3EvcuACCqCVm56oPNyGFXWir6o"}, "emitted_at": 1698881780826}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-10-06T01:31:54.742Z"}}, "data": {"payments": {"created_at": "2023-10-06T01:31:54.742Z"}}}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "created_at": "2023-10-06T00:44:56.869Z", "updated_at": "2023-10-06T00:44:58.179Z", "amount_money": {"amount": 100, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T00:44:56.976Z", "captured_at": "2023-10-06T00:44:57.050Z"}}, "location_id": "L82360QY4MG3B", "order_id": "ocxhWW5TrHQEasZ06AgJEGXgJZSZY", "processing_fee": [{"effective_at": "2023-10-06T02:44:58.000Z", "type": "INITIAL", "amount_money": {"amount": 33, "currency": "USD"}}], "total_money": {"amount": 100, "currency": "USD"}, "approved_money": {"amount": 100, "currency": "USD"}, "receipt_number": "btMa", "receipt_url": "https://squareupsandbox.com/receipt/preview/btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T00:44:56.869Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "JrHROPUWA23bC4pAUKpfUHyjLS9dyzLHnyjZGdyq8pC6o"}, "emitted_at": 1698881782702}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "created_at": "2023-10-06T01:31:54.742Z", "updated_at": "2023-10-06T01:31:56.396Z", "amount_money": {"amount": 892657, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T01:31:54.853Z", "captured_at": "2023-10-06T01:31:54.925Z"}}, "location_id": "L82360QY4MG3B", "order_id": "KjOIV2mxV5WDn1htAgCwl7skTkKZY", "processing_fee": [{"effective_at": "2023-10-06T03:31:56.000Z", "type": "INITIAL", "amount_money": {"amount": 25917, "currency": "USD"}}], "customer_id": "7879697", "total_money": {"amount": 892657, "currency": "USD"}, "approved_money": {"amount": 892657, "currency": "USD"}, "receipt_number": "tmKJ", "receipt_url": "https://squareupsandbox.com/receipt/preview/tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T01:31:54.742Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "BOxzRrA8O6eECUPcr4Q3EvcuACCqCVm56oPNyGFXWir6o"}, "emitted_at": 1698881782720}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-10-06T01:31:54.742Z"}}, "data": {"payments": {"created_at": "2023-10-06T01:31:54.742Z"}}}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "created_at": "2023-10-06T00:44:56.869Z", "updated_at": "2023-10-06T00:44:58.179Z", "amount_money": {"amount": 100, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T00:44:56.976Z", "captured_at": "2023-10-06T00:44:57.050Z"}}, "location_id": "L82360QY4MG3B", "order_id": "ocxhWW5TrHQEasZ06AgJEGXgJZSZY", "processing_fee": [{"effective_at": "2023-10-06T02:44:58.000Z", "type": "INITIAL", "amount_money": {"amount": 33, "currency": "USD"}}], "total_money": {"amount": 100, "currency": "USD"}, "approved_money": {"amount": 100, "currency": "USD"}, "receipt_number": "btMa", "receipt_url": "https://squareupsandbox.com/receipt/preview/btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T00:44:56.869Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "JrHROPUWA23bC4pAUKpfUHyjLS9dyzLHnyjZGdyq8pC6o"}, "emitted_at": 1698881783673}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "created_at": "2023-10-06T01:31:54.742Z", "updated_at": "2023-10-06T01:31:56.396Z", "amount_money": {"amount": 892657, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T01:31:54.853Z", "captured_at": "2023-10-06T01:31:54.925Z"}}, "location_id": "L82360QY4MG3B", "order_id": "KjOIV2mxV5WDn1htAgCwl7skTkKZY", "processing_fee": [{"effective_at": "2023-10-06T03:31:56.000Z", "type": "INITIAL", "amount_money": {"amount": 25917, "currency": "USD"}}], "customer_id": "7879697", "total_money": {"amount": 892657, "currency": "USD"}, "approved_money": {"amount": 892657, "currency": "USD"}, "receipt_number": "tmKJ", "receipt_url": "https://squareupsandbox.com/receipt/preview/tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T01:31:54.742Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "BOxzRrA8O6eECUPcr4Q3EvcuACCqCVm56oPNyGFXWir6o"}, "emitted_at": 1698881783699}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-10-06T01:31:54.742Z"}}, "data": {"payments": {"created_at": "2023-10-06T01:31:54.742Z"}}}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "created_at": "2023-10-06T00:44:56.869Z", "updated_at": "2023-10-06T00:44:58.179Z", "amount_money": {"amount": 100, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T00:44:56.976Z", "captured_at": "2023-10-06T00:44:57.050Z"}}, "location_id": "L82360QY4MG3B", "order_id": "ocxhWW5TrHQEasZ06AgJEGXgJZSZY", "processing_fee": [{"effective_at": "2023-10-06T02:44:58.000Z", "type": "INITIAL", "amount_money": {"amount": 33, "currency": "USD"}}], "total_money": {"amount": 100, "currency": "USD"}, "approved_money": {"amount": 100, "currency": "USD"}, "receipt_number": "btMa", "receipt_url": "https://squareupsandbox.com/receipt/preview/btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T00:44:56.869Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "JrHROPUWA23bC4pAUKpfUHyjLS9dyzLHnyjZGdyq8pC6o"}, "emitted_at": 1698881785512}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "created_at": "2023-10-06T01:31:54.742Z", "updated_at": "2023-10-06T01:31:56.396Z", "amount_money": {"amount": 892657, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T01:31:54.853Z", "captured_at": "2023-10-06T01:31:54.925Z"}}, "location_id": "L82360QY4MG3B", "order_id": "KjOIV2mxV5WDn1htAgCwl7skTkKZY", "processing_fee": [{"effective_at": "2023-10-06T03:31:56.000Z", "type": "INITIAL", "amount_money": {"amount": 25917, "currency": "USD"}}], "customer_id": "7879697", "total_money": {"amount": 892657, "currency": "USD"}, "approved_money": {"amount": 892657, "currency": "USD"}, "receipt_number": "tmKJ", "receipt_url": "https://squareupsandbox.com/receipt/preview/tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T01:31:54.742Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "BOxzRrA8O6eECUPcr4Q3EvcuACCqCVm56oPNyGFXWir6o"}, "emitted_at": 1698881785532}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-10-28T23:59:59.999999Z"}}, "data": {"payments": {"created_at": "2023-10-28T23:59:59.999999Z"}}}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-11-01T23:36:17.162702Z"}}, "data": {"payments": {"created_at": "2023-11-01T23:36:17.162702Z"}}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 8 records from payments stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as STOPPED"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881786339.9302, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "COMPLETE"}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing payments"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceSquare runtimes:\nSyncing stream payments 0:00:09.181622"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing SourceSquare"}}
```

The fixed version

```{"type": "LOG", "log": {"level": "INFO", "message": "Starting syncing SourceSquare"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as STARTED"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881570578.168, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "STARTED"}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: payments "}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-07-30T23:59:59.999999Z"}}, "data": {"payments": {"created_at": "2023-07-30T23:59:59.999999Z"}}}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-08-29T23:59:59.999999Z"}}, "data": {"payments": {"created_at": "2023-08-29T23:59:59.999999Z"}}}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-09-28T23:59:59.999999Z"}}, "data": {"payments": {"created_at": "2023-09-28T23:59:59.999999Z"}}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as RUNNING"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881574967.156, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "RUNNING"}}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "created_at": "2023-10-06T00:44:56.869Z", "updated_at": "2023-10-06T00:44:58.179Z", "amount_money": {"amount": 100, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T00:44:56.976Z", "captured_at": "2023-10-06T00:44:57.050Z"}}, "location_id": "L82360QY4MG3B", "order_id": "ocxhWW5TrHQEasZ06AgJEGXgJZSZY", "processing_fee": [{"effective_at": "2023-10-06T02:44:58.000Z", "type": "INITIAL", "amount_money": {"amount": 33, "currency": "USD"}}], "total_money": {"amount": 100, "currency": "USD"}, "approved_money": {"amount": 100, "currency": "USD"}, "receipt_number": "btMa", "receipt_url": "https://squareupsandbox.com/receipt/preview/btMaYRzB2vXFA0XT0G1ce1oQgNbZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T00:44:56.869Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "JrHROPUWA23bC4pAUKpfUHyjLS9dyzLHnyjZGdyq8pC6o"}, "emitted_at": 1698881574966}}
{"type": "RECORD", "record": {"stream": "payments", "data": {"id": "tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "created_at": "2023-10-06T01:31:54.742Z", "updated_at": "2023-10-06T01:31:56.396Z", "amount_money": {"amount": 892657, "currency": "USD"}, "status": "COMPLETED", "delay_duration": "PT168H", "source_type": "CARD", "card_details": {"status": "CAPTURED", "card": {"card_brand": "VISA", "last_4": "5858", "exp_month": 10, "exp_year": 2025, "fingerprint": "sq-1-3vwN-o-XjkorwFPBipqSiLc55lGE6E4-hHedFuQOS57Fa4MZUJPAGErnkTAH5MPgHw", "card_type": "CREDIT", "prepaid_type": "NOT_PREPAID", "bin": "453275"}, "entry_method": "KEYED", "cvv_status": "CVV_ACCEPTED", "avs_status": "AVS_ACCEPTED", "statement_description": "SQ *DEFAULT TEST ACCOUNT", "card_payment_timeline": {"authorized_at": "2023-10-06T01:31:54.853Z", "captured_at": "2023-10-06T01:31:54.925Z"}}, "location_id": "L82360QY4MG3B", "order_id": "KjOIV2mxV5WDn1htAgCwl7skTkKZY", "processing_fee": [{"effective_at": "2023-10-06T03:31:56.000Z", "type": "INITIAL", "amount_money": {"amount": 25917, "currency": "USD"}}], "customer_id": "7879697", "total_money": {"amount": 892657, "currency": "USD"}, "approved_money": {"amount": 892657, "currency": "USD"}, "receipt_number": "tmKJ", "receipt_url": "https://squareupsandbox.com/receipt/preview/tmKJkAUbg6ekBql1z7gRm8Ocj3EZY", "delay_action": "CANCEL", "delayed_until": "2023-10-13T01:31:54.742Z", "application_details": {"square_product": "ECOMMERCE_API", "application_id": "sandbox-sq0idb-7rb0cAiRI3eC5sPUvM8T6Q"}, "version_token": "BOxzRrA8O6eECUPcr4Q3EvcuACCqCVm56oPNyGFXWir6o"}, "emitted_at": 1698881574986}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-10-28T23:59:59.999999Z"}}, "data": {"payments": {"created_at": "2023-10-28T23:59:59.999999Z"}}}}
{"type": "STATE", "state": {"type": "STREAM", "stream": {"stream_descriptor": {"name": "payments"}, "stream_state": {"created_at": "2023-11-01T23:32:50.580243Z"}}, "data": {"payments": {"created_at": "2023-11-01T23:32:50.580243Z"}}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 2 records from payments stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Marking stream payments as STOPPED"}}
{"type": "TRACE", "trace": {"type": "STREAM_STATUS", "emitted_at": 1698881576181.821, "stream_status": {"stream_descriptor": {"name": "payments", "namespace": null}, "status": "COMPLETE"}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing payments"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceSquare runtimes:\nSyncing stream payments 0:00:05.604354"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing SourceSquare"}}```

**Review by marcosmarxm — APPROVED** [review]
Confirm the new code removes the duplicated ids for payments and refunds.

**Pull request merged**